### DISCUSSION TITLE
Best Approach to navigate between pages and setting state in StateNotifier

### DISCUSSION THREAD

**Discussion by dyardy** [rrousselGit/riverpod#1913] [Q&A]
Typical scenario, I have an application with a list of contacts (for example).  On this page the user can tap a contact and be redirected to the edit page for this contact.

I am using Riverpod/GoRouter.

The landing page (Edit) page after the click is used to view contact information and modify/save information.  The 'Edit' page is using a StateNotifier/StateNotifierProvider to maintain contact information (and calling the repository for saving).

What is the best approach to 
a) navigate from list page to edit page (passing the contact id)
b) how best to get the initial state (contact information) set in the stateNotifier? 
        i.e. #1 do i set the state in the provider to the active contact in the list page, then just navigate to the edit page? (**I do this now)**
              #2 do i pass in the id of the contact to the edit page which then creates notifierprovider passing in the id field which calls statenotifier with the initial parameter
              (I have had a problem with #2 approach, as any reference within the edit page has to always pass in some id field, even ref.watch calls)
              #3 other options??
              
              
              

I have come up against this error, which made to to start thinking there has got to be a better approach.
```
Tried to modify a provider while the widget tree was building.
If you are encountering this error, chances are you tried to modify a provider
in a widget life-cycle, such as but not limited to:
- build
- initState
- dispose
- didUpdateWidget
- didChangeDepedencies
```

**Comment by scopendo** [comment]
You might be interested in this approach https://github.com/rrousselGit/riverpod/discussions/1195

Note that this hasn't been revisited for Riverpod 2.x yet – I'm hoping that the new `overrideWith` method might make it easier.
